drivers/frequency/ad9545/ad9545.c

```diff
@@ -0,0 +1,2434 @@
+		init[i].hw_ch_num = i;
+		init[i].dev_desc = dev;
+
+		ret = no_os_clk_init(&tdc->hw, &init[i]);
+		if (ret < 0)
+			return ret;
+
+		dev->clks[AD9545_CLK_AUX_TDC][i] = tdc->hw;
+	}
+
+	return 0;
+}
+
+static int ad9545_aux_dpll_setup(struct ad9545_dev *dev)
+{
+	struct ad9545_aux_dpll_clk *clk;
+	struct no_os_clk_init_param init;
+	uint16_t regval;
+	int ret;
+	uint8_t val, i;
+
+	clk = &dev->aux_dpll_clk;
+	if (!clk->dpll_used)
+		return 0;
+
+	memset(&init, 0x00, sizeof(struct no_os_clk_init_param));
+
+	if (clk->source < NO_OS_ARRAY_SIZE(ad9545_in_clk_names)) {
+		val = clk->source;
+		clk->parent_clk = dev->ref_in_clks[val].hw;
+	} else {
+		val = clk->source - NO_OS_ARRAY_SIZE(ad9545_in_clk_names);
+		if (val > NO_OS_ARRAY_SIZE(ad9545_aux_tdc_clk_names))
+			return -EINVAL;
+		clk->parent_clk = dev->ref_in_clks[val].hw;
+		val = val + 0x6;
+	}
+
+	ret = ad9545_write_reg(dev, AD9545_AUX_DPLL_SOURCE, val);
+	if (ret < 0)
+		return ret;
+
+	/* write loop bandwidth in dHz */
+	regval = clk->loop_bw_mhz / 100;
+	ret = ad9545_write_reg_multiple(dev, AD9545_AUX_DPLL_LOOP_BW,
+					(uint8_t *)&regval, 2);
+	if (ret < 0)
+		return ret;
+
+	val = 0;
+	for (i = 0; i < NO_OS_ARRAY_SIZE(ad9545_rate_change_limit_map); i++) {
+		if (ad9545_rate_change_limit_map[i] == clk->rate_change_limit) {
+			val = i;
+			break;
+		}
+	}
+
+	ret = ad9545_write_reg(dev, AD9545_AUX_DPLL_CHANGE_LIMIT, val);
+	if (ret < 0)
+		return ret;
+
+	/* add compensation destination */
+	ret = ad9545_write_reg(dev, AD9545_COMPENSATE_DPLL,
+			       AD9545_COMPNESATE_VIA_AUX_DPLL);
+	if (ret < 0)
+		return ret;
+
+	ret = ad9545_write_reg(dev, AD9545_COMPENSATE_NCOS,
+			       AD9545_COMPENSATE_NCOS_VIA_AUX_DPLL);
+	if (ret < 0)
+		return ret;
+
+	ret = ad9545_write_reg(dev, AD9545_COMPENSATE_TDCS,
+			       AD9545_COMPENSATE_TDCS_VIA_AUX_DPLL);
+	if (ret < 0)
+		return ret;
+
+	init.name = ad9545_aux_dpll_name;
+	init.platform_ops = &ad9545_aux_dpll_clk_ops;
+	init.hw_ch_num = 1;
+	init.dev_desc = dev;
+
+	return no_os_clk_init(&clk->hw, &init);
+}
+
+
+static int ad9545_calib_apll(struct ad9545_dev *dev, int i)
+{
+	int cal_count;
+	uint8_t reg;
+	int ret;
+
+	/* APLL VCO calibration operation */
+	cal_count = 0;
+	while (cal_count < 2) {
+		ret = ad9545_write_reg(dev, AD9545_PWR_CALIB_CHX(i), 0);
+		if (ret < 0)
+			return ret;
+
+		ret = ad9545_io_update(dev);
+		if (ret < 0)
+			return ret;
+
+		ret = ad9545_write_reg(dev, AD9545_PWR_CALIB_CHX(i),
+				       AD9545_CALIB_APLL);
+		if (ret < 0)
+			return ret;
+
+		ret = ad9545_io_update(dev);
+		if (ret < 0)
+			return ret;
+
+		cal_count += 1;
+		no_os_mdelay(100);
+
+		ret = ad9545_read_reg(dev, AD9545_PLLX_STATUS(i), &reg);
+		if (ret < 0)
+			return ret;
+
+		if (AD9545_APLL_LOCKED(reg)) {
+			ret = ad9545_write_reg(dev, AD9545_PWR_CALIB_CHX(i), 0);
+			if (ret < 0)
+				return ret;
+
+			ret = ad9545_io_update(dev);
+			if (ret < 0)
+				return ret;
+
+			cal_count = 2;
+			break;
```

Comment:
Is break required here?

Reply:
In this case it seems redundant with the previous line.